### PR TITLE
Use Action.Prog.t to represent tools

### PR DESCRIPTION
**Pull request by rgrinberg**
This is ultimately the value that we need to pass to Command to create
build rules that fail as late as possible.

### CODE REVIEW

File: src/dune/context.mli [hunk 1]
```diff
@@ -73,10 +73,10 @@ type t = private
   ; ocaml_bin : Path.t
   ; ocaml : Path.t
   ; ocamlc : Path.t
-  ; ocamlopt : Path.t option
-  ; ocamldep : Path.t
-  ; ocamlmklib : Path.t
-  ; ocamlobjinfo : Path.t option  (** Environment variables *)
```
Comment (snowleopard) [inline src/dune/context.mli]: I guess the `(** Environment variables *)` comment was attached to `env : Env.t`.

**Comment by aalekseyev** [comment]
The change looks reasonable to me. Maybe it's worth a changelog entry? (and perhaps a test?)

**Review by snowleopard — APPROVED** [review]
Looks good to me!

**Comment by rgrinberg** [comment]
Indeed, but I think the comment is useless anyway.

**Comment by rgrinberg** [comment]
Let me give a try. It’s a bit hard to setup a context where these tools are absent.

**Comment by rgrinberg** [comment]
It's a bit hard to add a test, I've tested it manually and it seems to work fine.

**Pull request merged**